api_app/auth.py

```diff
@@ -0,0 +1,128 @@
+import logging
+
+from django.contrib.auth import authenticate, login, logout
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.exceptions import APIException
+from rest_framework_simplejwt.tokens import RefreshToken
+from rest_framework_simplejwt.views import TokenRefreshView
+from rest_framework.decorators import (
+    api_view,
+    authentication_classes,
+    permission_classes,
+)
+
+from api_app import serializers
+
+
+logger = logging.getLogger(__name__)
+
+
+""" Auth API endpoints """
+
+
+@api_view(["POST"])
+@authentication_classes([])
+@permission_classes([])
+def obtain_user_token(request):
+    """
+    REST endpoint to obtain user auth token via authentication
+
+    :param username: string
+        username of registered user
+    :param password: string
+        password of registered user
+
+    :return 202:
+        if accepted
+    :return 404:
+        if failed
+    """
+    try:
+        username = request.data["username"]
+        logger.info(f"obtain_user_token received request for '{username}'.")
+        password = request.data["password"]
+        auth_user = authenticate(username=username, password=password)
+        if auth_user:
+            # ---- Todo ----
```

Comment:
This change is yet to finalize.

Reply:
for me this is the best for the admin user experience